test/test_kubeclient.rb

```diff
@@ -232,6 +232,21 @@ def test_get_all
     assert_instance_of(Kubeclient::Namespace, result['namespace'][0])
   end
 
+  def test_api_bearer_token_with_params_success
+    stub_request(:get, 'http://localhost:8080/api/v1beta3/pods?labelSelector=name=redis-master')
+      .with(headers: { Authorization: 'Bearer valid_token' })
+      .to_return(body: open_test_json_file('pod_list_b3.json'),
+                 status: 200)
+
+    client = Kubeclient::Client.new 'http://localhost:8080/api/'
+    client.bearer_token('valid_token')
+
+    pods = client.get_pods(label_selector: 'name=redis-master')
+
+    assert_equal('Pod', pods.kind)
+    assert_equal(1, pods.size)
```

Comment:
please add assert_requested that indeed the request looks exactly as expected (and executed once, while at it :) )